[EDU-4242] feat: add app modernization archi (#985)
aziontech/docs: src/content/docs/en/pages/architectures/edge-application/application-modernization.mdx

---
title: Accelerate Application Modernization at the Edge

> description: Discover how Azion's Edge Platform enables application modernization through a serverless approach and microservices architecture. Learn how to migrate legacy applications to the edge, improving speed, reliability, and cost efficiency.

Reviewer: ```suggestion
description: Discover how Azion Edge Platform enables application modernization through a serverless approach and microservices architecture.
```

2. The request goes through Edge Firewall and is analyzed. If it's not a malicious request, it reaches the Edge Application.

3. The request reaches the edge application and based on its Rules Engine configuration, accesses the requested content. If configured, Image Processor, Edge Cache, Application Accelerator, and Edge Functions will be executed. 
With Edge Functions, you can have the logic performed at the edge, improving the usability and maintainability of decentralized applications. These functions are written in Javascript and run on Azion Edge Runtime.

4. The content, if static, will be stored in Edge Storage. So, it'll be requested and returned as a response.

5. Operations requiring database transactions can be accomplished with Edge SQL.

6. The application's logs are sent through Data Stream and can be analyzed by the customer.

7. If it's necessary to access an origin outside Azion:
    - Activating Origin Shield, your application is protected from a network list maintained by Azion.
    - Activating the Load Balancer module balances traffic across your origins, data centers, or cloud providers and mitigates network congestion and server overload.

>     - Auth Origin (?)

Reviewer: Validate this item

    - [Origins](/en/documentation/products/build/edge-application/origins/):  allows you to customize the source address of your Edge Application content, the Host header of your application, the path, and the HMAC authentication credentials for protected services.
    - [Domains](/en/documentation/products/build/edge-application/domains/): register a custom domain with Azion to deliver the edge application.
- [Azion Marketplace](/en/documentation/products/marketplace/) is a digital catalog that makes it easy to find, test, and deploy edge-running software.
- [Azion CLI](/en/documentation/products/azion-cli/overview/):  an open-source command-line interface (CLI) that lets you interact with Azion Edge Platform using a terminal. 
- [Azion Console](/en/documentation/products/guides/getting-to-know-azion-console/): Azion’s configuration interface where you can access all the available products and their settings. It enables you to complete all tasks in Azion’s platform

---

## Implementation

1. Create an edge application:

>     - On [Console](https://console.azion.com), create an application using the + Create button on the homepage.

Reviewer: ```suggestion
    - On [Console](https://console.azion.com), create an application using the **+ Create** button on the homepage.
```

    - Use the [Azion API](https://api.azion.com/#d43c5509-fb99-4b20-a7bd-64f0924af352) to create an edge application.
    - Use the [Azion CLI](/en/documentation/devtools/cli/create/#edge-applications) to create and deploy an edge application.
2. Create a domain and associate it with the edge application: 
    - On Console, [create a domain](/en/documentation/products/guides/configure-a-domain/) using the + Create button on the homepage.
    - Use the [Azion API](https://api.azion.com/#d43c5509-fb99-4b20-a7bd-64f0924af352) to create a domain.
    - If you created an application through a template, the domain is created for you.
3. Point domain to Azion: [update your DNS settings](/en/documentation/products/guides/migrate-ns-to-azion/) to point your domain's CNAME record to the Azion CDN endpoint provided for your domain. This step isn't mandatory.

:::note 
The edge application is accessible through the domain provided by Azion, which looks similar to: `ftd1cvntfl.map.azionedge.net`. 
:::

> 

Reviewer: ```suggestion
```

4. Test and monitor your edge application.

---

## Related docs

> - [Azion Marketplace](/en/documentation/products/marketplace/).
> - [Azion Edge Runtime Web APIs compatibility](/en/documentation/products/devtools/azion-edge-runtime/frameworks-compatibility/).
> - [Azion Edge Runtime Node.js compatibility](/en/documentation/products/azion-edge-runtime/compatibility/node/).

Reviewer: ```suggestion
- [Azion Marketplace](/en/documentation/products/marketplace/)
- [Azion Edge Runtime Web APIs compatibility](/en/documentation/products/devtools/azion-edge-runtime/frameworks-compatibility/)
- [Azion Edge Runtime Node.js compatibility](/en/documentation/products/azion-edge-runtime/compatibility/node/)
```
